tests/std/tests/P0896R4_ranges_alg_transform/test.cpp

```diff
@@ -0,0 +1,150 @@
+            basic_borrowed_range wrapped_input{input};
+            auto result = transform(
+                wrapped_input.begin(), wrapped_input.end(), basic_borrowed_range{output}.begin(), minus_one, get_first);
+            STATIC_ASSERT(same_as<decltype(result), unary_transform_result<I, O>>);
+            assert(result.in == wrapped_input.end());
+            assert(result.out == basic_borrowed_range{output}.end());
+            assert(ranges::equal(output, expected));
+        }
+    }
+
+    // Dual Range
+    {
+        const std::array<P, 3> input1      = {{{1, 99}, {4, 98}, {5, 97}}};
+        const std::array<double, 3> input2 = {{6.4, 7.9, 8.2}};
+        const std::array<int, 3> expected2 = {{7, 11, 13}};
+        using I1                           = iterator_t<basic_borrowed_range<P const>>;
+        using I2                           = iterator_t<basic_borrowed_range<double const>>;
+        using O                            = iterator_t<basic_borrowed_range<int>>;
+        constexpr auto cast                = [](double x) -> int { return static_cast<int>(x); };
+
+        { // Validate range overload
+            std::array<int, 3> output{};
+            auto result = transform(basic_borrowed_range{input1}, basic_borrowed_range{input2},
+                basic_borrowed_range{output}.begin(), std::plus<int>{}, get_first, cast);
+            STATIC_ASSERT(same_as<decltype(result), binary_transform_result<I1, I2, O>>);
+            assert(result.in1 == basic_borrowed_range{input1}.end());
+            assert(result.in2 == basic_borrowed_range{input2}.end());
+            assert(result.out == basic_borrowed_range{output}.end());
+            assert(ranges::equal(output, expected2));
+        }
+        { // Validate iterator + sentinel overload
+            std::array<int, 3> output{};
+            basic_borrowed_range wrapped_input1{input1};
+            basic_borrowed_range wrapped_input2{input2};
+            auto result = transform(wrapped_input1.begin(), wrapped_input1.end(), wrapped_input2.begin(),
+                wrapped_input2.end(), basic_borrowed_range{output}.begin(), std::plus<int>{}, get_first, cast);
+            STATIC_ASSERT(same_as<decltype(result), binary_transform_result<I1, I2, O>>);
+            assert(result.in1 == wrapped_input1.end());
+            assert(result.in2 == wrapped_input2.end());
+            assert(result.out == basic_borrowed_range{output}.end());
+            assert(ranges::equal(output, expected2));
+        }
+    }
+}
+
+int main() {
+    STATIC_ASSERT((smoke_test(), true));
+    smoke_test();
+}
+
+struct unary_instantiator {
+    template <class In, class Out>
+    static void call(In&& in = {}, Out out = {}) {
+        using ranges::begin, ranges::transform, ranges::end, ranges::iterator_t;
+
+        constexpr UnaryFunctorFor<iterator_t<In>, Out> func{};
+        constexpr ProjectedUnaryFunctorFor<Out> projfunc{};
+        constexpr ProjectionFor<iterator_t<In>> proj{};

```

Comment:
These are why your test is failing: `func` returns something, but that thing is most likely not writable to `out`. (These aliases are semi-abandoned since the `test::range` rework - I plan to eliminate them once I find the time to rewrite the few "old" tests that use them.)

Reply:
So lucky me took the wrong test as template

Reply:
Removed